pkg/api/v1alpha1/cluster_types.go

```diff
@@ -832,11 +832,48 @@ var validCiliumPolicyEnforcementModes = map[CiliumPolicyEnforcementMode]bool{
 	CiliumPolicyModeNever:   true,
 }
 
+// Reasons for the terminal failures while reconciling the Cluster object.
+const (
+	// MissingDependentObjectsReason reports that the Cluster is missing dependent objects.
+	MissingDependentObjectsReason = "MissingDependentObjects"
+
+	// ManagementClusterRefInvalidReason reports that the Cluster management cluster reference is invalid. This
+	// can whether if it does not exist or the cluster referenced is not a management cluster.
+	ManagementClusterRefInvalidReason = "ManagementClusterRefInvalid"
+
+	// ClusterSpecValidationFailedReason reports that the Cluster spec validation has failed.
+	ClusterSpecValidationFailedReason = "ClusterSpecValidationFailed"
+
+	// DatacenterConfigValidationFailedReason reports that the Cluster datacenterconfig validation has failed.
+	DatacenterConfigValidationFailedReason = "DatacenterConfigValidationFailed"
+
+	// MachineConfigValidationFailedReason reports that the Cluster machineconfig validation has failed.
+	MachineConfigValidationFailedReason = "MachineConfigValidationFailed"
+
+	// UnavailableControlPlaneIPReason reports that the Cluster controlPlaneIP is already in use.
+	UnavailableControlPlaneIPReason = "UnavailableControlPlaneIP"
+)
+
+// Reasons for the terminal failures while reconciling the Cluster object specific for Tinkerbell.
+const (
+	// HardwareValidationFailedReason reports that the hardware validation has failed.
+	HardwareValidationFailedReason = "HardwareValidationFailed"
+
+	// RufioMachineValidationFailedReason reports that the rufio machine validation has failed.
+	RufioMachineValidationFailedReason = "RufioMachineValidationFailed"
+)
+
 // ClusterStatus defines the observed state of Cluster.
 type ClusterStatus struct {
 	// Descriptive message about a fatal problem while reconciling a cluster
 	// +optional
 	FailureMessage *string `json:"failureMessage,omitempty"`
+
+	// Machine readable value about a terminal problem while reconciling the cluster
+	// set at the same time as failureMessage
+	// +optional
+	FailureReason *string `json:"failureReason,omitempty"`
```

Comment:
maybe worth making this its own type? it sets intention about these field having a limited number of preset values.

Reply:
Good idea